extensions/amp-story/1.0/progress-bar.js

```diff
@@ -119,10 +180,191 @@ export class ProgressBar {
+   * @param {boolean} shouldAnimate
+   * @private
+   */
+  render_(shouldAnimate = true) {
+    const segmentWidth = this.getSegmentWidth_();
+    let translateX =
+      -(this.firstExpandedSegmentIndex_ - this.getPrevEllipsisCount_()) *
+      (ELLIPSE_WIDTH_PX + SEGMENTS_MARGIN_PX);
+
+    this.resources_
+      .mutateElement(this.root_, () => {
+        for (let index = 0; index < this.segmentCount_; index++) {
+          const width =
+            index >= this.firstExpandedSegmentIndex_ &&
+            index < this.firstExpandedSegmentIndex_ + MAX_SEGMENTS
+              ? segmentWidth
+              : ELLIPSE_WIDTH_PX;
+          this.transform_(this.segments_[index], translateX, width);
+          translateX += width + SEGMENTS_MARGIN_PX;
+        }
+      })
+      .then(() => {
+        this.root_.classList.toggle(
+          'i-amphtml-animate-progress',
+          shouldAnimate
+        );
+      });
+  }
+
+  /**
+   * Applies transform to a segment.
+   * @param {!Element} segment
+   * @param {number} translateX
+   * @param {number} width
+   * @private
+   */
+  transform_(segment, translateX, width) {
+    if (this.storeService_.get(StateProperty.RTL_STATE)) {
+      translateX *= -1;
+    }
+
+    // Do not remove translateZ(0.00001px) as it prevents an iOS repaint issue.
+    // http://mir.aculo.us/2011/12/07/the-case-of-the-disappearing-element/
+    segment.setAttribute(
+      'style',
+      `transform: translate3d(${translateX}px, 0px, 0.00001px) scaleX(${width /
+        ELLIPSE_WIDTH_PX});`
+    );
+  }
+
+  /**
+   * Gets the individual segment width.
+   * @return {number}
+   * @private
+   */
+  getSegmentWidth_() {
+    const nextEllipsisCount = this.getNextEllipsisCount_();
+    const prevEllipsisCount = this.getPrevEllipsisCount_();
+    const totalEllipsisWidth =
+      (nextEllipsisCount + prevEllipsisCount) *
+      (ELLIPSE_WIDTH_PX + SEGMENTS_MARGIN_PX);
+    const totalSegmentsWidth = this.barWidthPx_ - totalEllipsisWidth;
+
+    return (
+      totalSegmentsWidth / Math.min(this.segmentCount_, MAX_SEGMENTS) -
+      SEGMENTS_MARGIN_PX
+    );
+  }
+
+  /**
+   * Gets the number of ellipsis that should appear to the right (in LTR) of the
+   * expanded segments.
+   * @return {number}
+   * @private
+   */
+  getNextEllipsisCount_() {
+    const nextPagesCount =
+      this.segmentCount_ - (this.firstExpandedSegmentIndex_ + MAX_SEGMENTS);
+    return nextPagesCount > 3 ? 3 : Math.max(nextPagesCount, 0);
+  }
+
+  /**
+   * Gets the number of ellipsis that should appear to the left (in LTR) of the
+   * expanded segments.
+   * @return {number}
+   * @private
+   */
+  getPrevEllipsisCount_() {
+    return Math.min(3, this.firstExpandedSegmentIndex_);
+  }
+
+  /**
+   * Checks if an index is past the MAX_SEGMENTS limit and updates the progress
+   * bar accordingly.
+   * @private
+   */
+  checkIndexForOverflow_() {
+    // Touching an ellipsis on the right (LTR).
```

Comment:
Plz update this comment and the following one. We don't need to talk about right or left but simply previous/next, which work for both RTL and LTR